### ISSUE TITLE
svg -> png broken with newer inkscape version?

### ISSUE THREAD

**Issue by fschwa** [fabiensanglard/gebbdoom#31]
Hi,

I was not able to generate the PDF file with all images. All SVG images were missing. While investigating, I found that the inkscape command given somehow did not work.

Replacing it with the command scheme `inkscape --export-filename=out.png --export-dpi=300 in.svg` worked.

```
$ inkscape --version
Inkscape 1.0.2 (e86c870879, 2021-01-15)
    Pango version: 1.48.1
```

Not sure how to fix the version detection line in make.sh.

**Comment by fabiensanglard** [comment]
Can you tell me what is the output of the command
```
which inskcape
```
Also what OS are you using?

**Comment by fschwa** [comment]
```
$ which inkscape
/usr/bin/inkscape
```

This is on Arch Linux.

**Comment by fabiensanglard** [comment]
That is the path used in make.sh. What is the exact error message you are getting?

**Comment by fschwa** [comment]
Hi,

when I run `make.sh`, there is no error message. So I edited the `make.sh` to remove output redirect:
```
-               inkscape $exportOption="$dst" --without-gui --export-dpi=100 "$src" > /dev/null 2>&1
+               inkscape $exportOption="$dst" --without-gui --export-dpi=100 "$src"
```
So then it outputs stuff like this:
```
$ ./make.sh
    Pango version: 1.48.2
Convert screenshots_svg/3do_motherboard.svg to PNG (100dpi).
Warning: Option --without-gui= is deprecated
Unbekannte Option --export-file=/home/fs/git/gebbdoom/src/screenshots/3do_motherboard.png
Convert screenshots_svg/3do_vectors.svg to PNG (100dpi).
Warning: Option --without-gui= is deprecated
Unbekannte Option --export-file=/home/fs/git/gebbdoom/src/screenshots/3do_vectors.png
Convert screenshots_svg/486_layout.svg to PNG (100dpi).
Warning: Option --without-gui= is deprecated
Unbekannte Option --export-file=/home/fs/git/gebbdoom/src/screenshots/486_layout.png
```
with `Unbekannte Option` being `unknown option`.

Changing the inkscape command to the one given in the initial post works without a problem with my version of Inkscape.

Hope that helps a bit.

**Comment by fschwa** [comment]
> That is the path used in make.sh. What is the exact error message you are getting?

Not sure I understand this remark. In the `make.sh` here there is no path used at all. It's just running plain `inkscake` here, so looking it up in the `$PATH` environment variable. Or am I missing something?

**Comment by fabiensanglard** [comment]
The script attempts to detect what version of inkscape is running and uses either --export-file or --export-png.
In your case it seems the outcome of this line 
```
[[ $(inkscape --version | awk -F '[ .]' '{ print $2 }') -gt 0 ]] && exportOption="--export-file" || exportOption="--export-png"
```
If not doing what is expected.

What value does the following command returns

```
inkscape --version | awk -F '[ .]' '{ print $2 }'
```

**Comment by deluxe** [comment]
Inkscape seems to have deprecated command `--export-file` since 1.0rc1.
`--export-file` should replaced by `--export-filename`.

`--without-gui` is an old option, already deprecated in 0.92. It can be removed without impact.

**Comment by fabiensanglard** [comment]
Probably no the right thing to support older versions of inkscape. If we do, make.sh will become a mess.

The fix would be to request a minimum version of inkscape in make.sh, fail if it is not there, and use the appropriate flags.

**Comment by deluxe** [comment]
Dropping support for older versions of Inkscape is sensible in the long run. Now fixing this precise bug doesn't require adding any mess to make.sh. See for yourself:
1. Updating options and keeping the support (for now): https://github.com/deluxe/gebbdoom/commit/a5ff5b752172e5b13f989ab60cb413a02cef0ca0
2. Checking version and cutting support: https://github.com/deluxe/gebbdoom/commit/8a21e04361dc3238e33b8761e3826d091bab7cef

**Comment by fabiensanglard** [comment]
#1 Maybe trigger older version of Inkscape to start the GUI doesn't it?
#2 Looks like the way to go.

**Comment by fschwa** [comment]
```
$ inkscape --version | awk -F '[ .]' '{ print $2 }'
    Pango version: 1.48.2
1
```

**Comment by deluxe** [comment]
1. does not trigger GUI on version 0.92.x of Inkscape but may indeed trigger GUI on even older versions.
2. cut the support clearly.

I'll test 2. on Linux make you a Pull Request.

**Comment by deluxe** [comment]
Fixed by #32.
